cli/src/pcluster/config/validators.py

```diff
@@ -769,6 +769,41 @@ def s3_bucket_validator(param_key, param_value, pcluster_config):
     return errors, warnings
 
 
+def s3_bucket_region_validator(param_key, param_value, pcluster_config):
+    """Validate S3 bucket  is in the same region with the cloudformation stack."""
+    errors = []
+    warnings = []
+    s3_client = boto3.client("s3")
+    try:
+        bucket_region = s3_client.get_bucket_location(Bucket=param_value).get("LocationConstraint")
+        # Buckets in Region us-east-1 have a LocationConstraint of null
+        if bucket_region is None:
+            bucket_region = "us-east-1"
+        if bucket_region != pcluster_config.region:
```

Comment:
We need to consider the case that the region is provided through the CLI, i.e. `pcluster create <some_cluster> -r us-east-1`. We might need to validate this in `commands.py` or somehow pass in the region argument

Reply:
The case `-r us-east-1` is already considered here, since the pcluster_config in the validator stage already convert all the input arguments into the config file.

Reply:
The case -r us-east-1 is already considered here, since the pcluster_config in the validator stage already convert all the input arguments into the config file.